src/client/styles/button.scss

```diff
@@ -0,0 +1,37 @@
+/**
+ *
+ *  Online store PWA sample.
+ *  Copyright 2017 Google Inc. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ *
+ */
+
+ button {
+   background-color: var(--primary-color);
```

Comment:
Yay, custom properties!

In order to support old browsers, however, I think we should have defaults everywhere we use them. For example, if the default primary color we want would be `blue`, we should follow the pattern:

```css
background-color: blue;
background-color: var(--primary-color, blue);
```

Browsers that don't support custom properties would use the first definition, whereas browsers that do would use the second (and they would still have a default if the custom property value is missing). This is the most bullet-proof pattern I've found for gracefully degrading custom properties.

Oh, and we can make the default value a SASS variable if we want to add some DRYness to the code:

```scss
background-color: $primary-color;
background-color: var(--primary-color, $primary-color);
```

Reply:
Hm, I wonder - if we encode this in sass, what's the advantage of css variables on top of it? Maybe we should stick to sass only?

Reply:
Depends on what we want to do with them, really.

SASS variables have the advantage of being more expressive at compile-time; they can be used as part of selectors, and can be interpolated, for example. They also allow for interesting operations, like calculating darker shades or contrasting text colors automatically.

Custom properties have the advantage of fitting into the cascade model and being runtime entities. That's particularly important in custom elements, where you might want to use them as a way of overriding styles or providing parts of the style through a controlled API (rather than forcing the developer to learn the internals of your component).

In the case of a button, the internal structure is simple enough that if a user wants to override its colors, they can just target it directly in CSS, which means we probably wouldn't need custom properties. For something more complex, like e.g. a date selector, where the primary color might appear in different contexts as background, text, or border colors, it might be interesting to expose it as a custom property and give a developer the ability of changing it all in one go (if they're ok with limiting browser support).

So, all in all, considering what this PR is about, feel free to stick to SASS vars for now!

Reply:
To summarise offline discussion: 
+ if we support evergreen browsers only -> go with css vars
+ if we want a one-off code with older browsers support -> go with sass vars
+ if we want a library-like code with older browsers support -> go with css vars + sass duplication

Therefore, for this PR, we go with css vars (evergreen browsers) with accidental hardcoded fallback if needed for particular use case.